File: CS/Task26/MyDateEditCalendarPainter.cs
using System.Drawing;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Calendar;

namespace Task26
{
    public class MyDateEditCalendarPainter : DateEditPainter
    {
        public MyDateEditCalendarPainter(MyDateEditCalendar calendar)
            : base(calendar)
        {
        }

        public new MyDateEditCalendar Calendar
        {
            get { return base.Calendar as MyDateEditCalendar; }
        }

        public override void DrawDayCellText(DayNumberCellInfo cell, Brush brush)
        {
            base.DrawDayCellText(cell, brush);
            if (Calendar.Properties.HaveNote(cell.Date))
                DrawMarker(cell.Bounds, cell.Graphics);
        }

        protected virtual void DrawMarker(Rectangle Bounds, Graphics G)
        {
            int MarkerWidth = Bounds.Width / 4;
            Point[] Marker = {
                new Point(Bounds.Right - MarkerWidth, Bounds.Bottom),
                new Point(Bounds.Right, Bounds.Bottom - MarkerWidth),
                new Point(Bounds.Right, Bounds.Bottom)
            };
            G.DrawPolygon(Pens.Red, Marker);
        }
    }
}

File: VB/Task26/MyDateEditCalendarPainter.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Calendar

Namespace Task26
	Public Class MyDateEditCalendarPainter
		Inherits DateEditPainter
		Public Sub New(ByVal calendar As MyDateEditCalendar)
			MyBase.New(calendar)
		End Sub

		Public Shadows ReadOnly Property Calendar() As MyDateEditCalendar
			Get
				Return TryCast(MyBase.Calendar, MyDateEditCalendar)
			End Get
		End Property

		Public Overrides Sub DrawDayCellText(ByVal cell As DayNumberCellInfo, ByVal brush As Brush)
			MyBase.DrawDayCellText(cell, brush)
			If Calendar.Properties.HaveNote(cell.Date) Then
				DrawMarker(cell.Bounds, cell.Graphics)
			End If
		End Sub

		Protected Overridable Sub DrawMarker(ByVal Bounds As Rectangle, ByVal G As Graphics)
			Dim MarkerWidth As Integer = Bounds.Width / 4
			Dim Marker() As Point = { New Point(Bounds.Right - MarkerWidth, Bounds.Bottom), New Point(Bounds.Right, Bounds.Bottom - MarkerWidth), New Point(Bounds.Right, Bounds.Bottom) }
			G.DrawPolygon(Pens.Red, Marker)
		End Sub
	End Class
End Namespace
